Source/JavaScriptCore/bytecode/MetadataTable.cpp

```diff
@@ -61,16 +61,17 @@ MetadataTable::~MetadataTable()
 
 void MetadataTable::destroy(MetadataTable* table)
 {
-    RefPtr<UnlinkedMetadataTable> unlinkedMetadata = WTFMove(table->linkingData().unlinkedMetadata);
-
-    table->~MetadataTable();
-
     // FIXME: This check should really not be necessary, see https://webkit.org/b/272787
-    if (UNLIKELY(!unlinkedMetadata)) {
+    if (table->isDestroyed()) {
         ASSERT_NOT_REACHED();
         return;
     }
 
+    RefPtr<UnlinkedMetadataTable> unlinkedMetadata = WTFMove(table->linkingData().unlinkedMetadata);
```

Comment:
So, there is a case that the `MetadataTable` was valid however its `unlinkedMetadata` was gone. That's why we need to check `isDestroyed` first, here.

Reply:
`unlinkedMetadata` is exchanged with null here, and this should be the only place that it is set to null.  The pre-existing code was doing the null check after the swap out.  It was also calling `table->~MetadataTable()` regardless.  This doesn't make sense.  Destruction should only happen once.  If there's some way that we can get to this function more than once, then the 2nd time should not execute any of this code.  Hence, I moved the null check to the top.